exchanges/orderbook/orderbook_types.go

```diff
@@ -38,23 +38,23 @@ var (
 	errChecksumStringNotSet = errors.New("checksum string not set")
 )
 
-var service = Service{
-	books: make(map[string]Exchange),
-	Mux:   dispatch.GetNewMux(nil),
+var s = store{
+	orderbooks:  make(map[key.ExchangePairAsset]bookWithExchangeID),
+	exchangeIDs: make(map[string]uuid.UUID),
+	signalMux:   dispatch.GetNewMux(nil),
 }
 
-// Service provides a store for difference exchange orderbooks
-type Service struct {
-	books map[string]Exchange
-	*dispatch.Mux
-	mu sync.Mutex
+type bookWithExchangeID struct {
+	ExchangeID uuid.UUID
```

Comment:
🚧 Jump to comment on `store` below.
In this context, I don't think that it matters that the router belongs to an exchange. The book has a router to publish to, and that's all that matters.
```suggestion
	RouterID uuid.UUID
```